x-pack/plugins/data_visualizer/public/application/index_data_visualizer/utils/get_supported_aggs.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { DataViewField } from '@kbn/data-views-plugin/common';
+
+export const isCounterTimeSeriesMetricField = (field: DataViewField) =>
+  field.timeSeriesMetric === 'counter';
+
+export const isGaugeTimeSeriesMetricField = (field: DataViewField) =>
+  field.timeSeriesMetric === 'gauge';
+
+const SUPPORTED_AGGS = {
+  COUNTER: new Set([
+    'count',
+    'histogram',
+    'variable_width_histogram',
+    'rate',
+    'min',
+    'max',
+    'top_metrics',
+    'range',
+  ]),
+  GAUGE: new Set([
+    'count',
+    'max',
+    'top_metrics',
+    'missing',
+    'date_histogram',
+    'sum',
+    'rate',
+    'boxplot',
+    'value_count',
+    'avg',
+    'percentiles',
+    'cardinality',
+    'histogram',
+    'variable_width_histogram',
+    'frequent_item_sets',
+    'min',
+    'stats',
+    'diversified_sampler',
+    'percentile_ranks',
+    'median_absolute_deviation',
+    'multi_terms',
+    'auto_date_histogram',
+    'rare_terms',
+    'range',
+    'extended_stats',
+    'date_range',
+    'terms',
+    'significant_terms',
+  ]),
+  AGGREGATABLE: new Set(['count', 'cardinality', 'percentiles', 'stats', 'terms']),
```

Comment:
Is this list complete? It doesn't contain `avg` for example which means that opening Lens from any numeric field now uses `max` where it used to be `avg`.

Reply:
Fixed here [`413bb2a` (#153893)](https://github.com/elastic/kibana/pull/153893/commits/413bb2a4ab4fb92f3ba3da57302daa84766e7d9c)